opm/autodiff/FlowMain.hpp

```diff
@@ -129,6 +129,7 @@ namespace Opm
             asImpl().setupGridAndProps();
             asImpl().setupState();
             asImpl().distributeData();
+            asImpl().runDiagnostics();
```

Comment:
>I think it might make more sense to run the diagnostics immediately after setupGridAndProps() instead.

Does grid object created before this function? Since diagnostics need grid object.